lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -937,6 +938,44 @@ diagnoseCopyToHost(SILValue value, SILInstruction *user, SILLocation loc) {
 }
 
 
+void TFFunctionPartition::markPredTerm(TermInst *predTerm) {
+  switch (predTerm->getTermKind()) {
+  case TermKind::UnreachableInst:
+  case TermKind::ReturnInst:
+  case TermKind::ThrowInst:
+  case TermKind::UnwindInst: {
+    llvm_unreachable("Did not expect terminator with no successors!");
+    return;
+  }
+  case TermKind::BranchInst:
+  case TermKind::CondBranchInst: {
+    markInstruction(*predTerm, Marking::Copy);
+    return;
+  }
+  case TermKind::SwitchEnumInst:
+  case TermKind::SwitchEnumAddrInst:
+  case TermKind::TryApplyInst:
+  case TermKind::DynamicMethodBranchInst:
+  case TermKind::CheckedCastBranchInst:
+  case TermKind::CheckedCastAddrBranchInst: {
+    markedInstructions.insert({predTerm, Marking::Send});
+    return;
+  }
+  case TermKind::CheckedCastValueBranchInst:
+  case TermKind::SwitchValueInst:
```

Comment:
Does this inst make sense in the Swift for TensorFlow context?

From https://github.com/apple/swift/blob/master/docs/SIL.rst#switch-value, it seems user could write code like:
```swift
public func foo(x: Int) {
  switch(x) {
  case 0: <some tensor stuff>
  case 1: <some tensor stuff>
  default: ...
}
```

Reply:
We should support all valid SIL terminators in the partitioner as long as there are valid Swift programs that generate such terminators. 

> In fact, it is transformed to cond_br under -O

Is it simplified by SimplifyCFG? We can't turn of `-O` yet because TFDeabstraction is not fully finished. You could leave a TODO or NOTE in the test case.